compiler/ballerina-treegen/src/main/resources/syntax_tree_descriptor.json

```diff
@@ -3977,13 +3977,18 @@
             ]
         },
         {
-            "name": "FunctionalMatchPatternNode",
+            "name": "ErrorMatchPatternNode",
             "base": "Node",
-            "kind": "FUNCTIONAL_MATCH_PATTERN",
+            "kind": "ERROR_MATCH_PATTERN",
             "attributes": [
                 {
-                    "name": "typeRef",
-                    "type": "Node"
+                    "name": "errorKeyword",
+                    "type": "Token"
+                },
+                {
+                    "name": "typeReference",
+                    "type": "Node",
```

Comment:
Can the "type" be a `NameReference` node?

Reply:
changed